### PR TITLE
[Fiber] Component lifecycle tests

### PR DESCRIPTION
**Pull request by acdlite**
Fixes tests in ReactComponentLifeCycle-test.js

Biggest change is to set current owner to null at beginning of commit phase.

### CODE REVIEW

File: scripts/fiber/tests-failing.txt [hunk 1]
```diff
@@ -42,8 +42,8 @@ src/renderers/shared/hooks/__tests__/ReactComponentTreeHook-test.js
 src/renderers/shared/hooks/__tests__/ReactHostOperationHistoryHook-test.js
 * gets recorded during an update
 
-src/renderers/shared/shared/__tests__/ReactComponentLifeCycle-test.js
-* should carry through each of the phases of setup
+src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js
+* should batch unmounts
```
Comment (acdlite) [inline scripts/fiber/tests-failing.txt]: Oops just noticed this

Comment (bvaughn) [inline scripts/fiber/tests-failing.txt]: Guh. Sorry for missing it.

File: src/renderers/shared/fiber/ReactFiberCommitWork.js [hunk 2]
```diff
@@ -343,6 +344,7 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
         if (typeof instance.componentWillUnmount === 'function') {
           safelyCallComponentWillUnmount(current, instance);
         }
+        ReactInstanceMap.remove(instance);
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberCommitWork.js]: Never safe to rely on unmount.

File: src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js [hunk 3]
```diff
@@ -345,6 +349,15 @@ describe('ReactCompositeComponent-state', () => {
     expect(() => {
       ReactDOM.unmountComponentAtNode(container);
     }).not.toThrow();
+
+    // Stack doesn't warn
+    if (ReactDOMFeatureFlags.useFiber) {
```
Comment (gaearon) [inline src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js]: Why?

Also, if it doesn't warn, do you think this could create warning noise in codebases that already rely on this pattern? It would be nice to check Power Editor and other React-heavy projects for this.

Comment (acdlite) [inline src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js]: My hunch is that `setState` on a parent in `componentWillUnmount` isn't that common but yeah I'll check

**Comment by acdlite** [comment]
@sebmarkbage Updated

File: src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js (hunk 3 above)
Comment (acdlite) [inline src/renderers/shared/shared/__tests__/ReactCompositeComponentState-test.js]: Code changed, moot now anyway

File: src/renderers/shared/fiber/ReactFiberClassComponent.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ var invariant = require('invariant');
 const isArray = Array.isArray;
 
 module.exports = function(
-  scheduleUpdate : (fiber : Fiber, priorityLevel : PriorityLevel) => void,
+  scheduleUpdate : (fiber : Fiber, priorityLevel : PriorityLevel, callerName : string | null) => void,
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: I'm not a fan of this pattern since they don't get stripped in prod. Can we reword it to be `setState, replaceState or forceUpdate` in the message instead of complicating the code?

Comment (acdlite) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: I'm passing `null` in prod. But yeah, I like that better anyway.

Comment (acdlite) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: Should do the same in `ReactFiberUpdateQueue` https://github.com/facebook/react/blob/master/src/renderers/shared/fiber/ReactFiberUpdateQueue.js#L217

Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberClassComponent.js]: yea

**Comment by acdlite** [comment]
@sebmarkbage Updated again

**Comment by acdlite** [comment]
Forgot to update the Stack tests

**Comment by acdlite** [comment]
Fixed

File: src/renderers/dom/shared/findDOMNode.js [hunk 5]
```diff
@@ -26,18 +26,26 @@ let findStack = function(arg) {
...
-        (owner: any)._warnedAboutRefsInRender,
+        warnedAboutRefsInRender,
         '%s is accessing findDOMNode inside its render(). ' +
         'render() should be a pure function of props and state. It should ' +
         'never access something that requires stale data from the previous ' +
         'render, such as refs. Move this logic to componentDidMount and ' +
         'componentDidUpdate instead.',
         getComponentName(owner) || 'A component'
       );
-      (owner: any)._warnedAboutRefsInRender = true;
```
Comment (gaearon) [inline src/renderers/dom/shared/findDOMNode.js]: Oops. Should we preventExtensions on fibers? Like we do in Stack instantiate.

Comment (sebmarkbage) [inline src/renderers/dom/shared/findDOMNode.js]: Yea. Let's add it in the constructor in DEV.

Comment (sebmarkbage) [inline src/renderers/dom/shared/findDOMNode.js]: But expandos on the class isn't really better neither.
